src/Avalonia.Controls/Button.cs

```diff
@@ -304,15 +304,18 @@ protected override void OnKeyUp(KeyEventArgs e)
         /// </summary>
         protected virtual void OnClick()
         {
-            OpenFlyout();
+            if (IsEnabled & IsEffectivelyEnabled)
+            {
+                OpenFlyout();
 
-            var e = new RoutedEventArgs(ClickEvent);
-            RaiseEvent(e);
+                var e = new RoutedEventArgs(ClickEvent);
+                RaiseEvent(e);
 
-            if (!e.Handled && Command?.CanExecute(CommandParameter) == true)
-            {
-                Command.Execute(CommandParameter);
-                e.Handled = true;
+                if (!e.Handled && Command?.CanExecute(CommandParameter) == true)
+                {
+                    Command.Execute(CommandParameter);
+                    e.Handled = true;
+                }
             }
```

Comment:
Can you add tests for that? By raising key down event on the button while it's disabled (and another test when parent is disabled).
Also, I think it should be enough to use IsEffectivelyEnabled.

Reply:
I accidentally closed this pr while reverting the commits.
https://github.com/AvaloniaUI/Avalonia/pull/7206 This is the new pr